Library/Homebrew/utils.rb

```diff
@@ -337,6 +337,14 @@ def with_system_path
   ENV["PATH"] = old_path
 end
 
+def with_custom_locale(locale)
+  old_locale = ENV["LC_ALL"]
+  ENV["LC_ALL"] = locale
+  yield
+ensure
+  ENV["LC_ALL"] = old_locale
+end
```

Comment:
Let's stick with `with_custom_locale` for now. I kind of want to get it shipped & see whether we see an appreciable difference in bottles being relocatable, and these discussions have tended to get bogged down indefinitely on minor issues like this.

Happy to reconsider later, or even review a follow-up PR to introduce more generic handling throughout the codebase, but yeah, I'm relatively wary of this kind of discussion by now having seen it delay or outright derail core PRs for weeks on end too much over the last ~6 months 😕.